packages/frontend-v2/pages/signup.tsx

```diff
@@ -0,0 +1,115 @@
+import { NextPage } from "next";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { classValidatorResolver } from "@hookform/resolvers/class-validator";
+import { SignUpStudentDto } from "../temp/dto-types";
+import { API } from "@graduate/api-client";
+import {
+  Button,
+  FormControl,
+  FormErrorMessage,
+  Input,
+  InputGroup,
+  Text,
+} from "@chakra-ui/react";
+import { useRouter } from "next/router";
+import { AxiosError } from "axios";
+
+const Signup: NextPage = () => {
+  const [apiError, setApiError] = useState("");
+
+  const router = useRouter();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting },
+  } = useForm<SignUpStudentDto>({
+    resolver: classValidatorResolver(SignUpStudentDto),
```

Comment:
same comment as above, probably save complex validtion for server side

Reply:
the dtos don't necessarily do complex validation? they mostly just validate the shape of the json with some sanity constraints so i think it may be fine to do the validation client side especially since we don't need any additional logic

Reply:
will the backend not need to re-do the validation anyway? i think pulling in an extra library and dependency is not worth it

Reply:
fair, good with having class validator remove from the frontend